x/concentrated-liquidity/swaps.go

```diff
@@ -23,34 +23,41 @@ type SwapState struct {
 	liquidity                sdk.Dec // new liquidity when swap is done
 }
 
+// CreateNewConcentratedLiquidityPool creates a new concentrated liquidity pool with the given parameters.
+// The pool tokens are denom0 and denom1, and are ordered such that denom0 is lexicographically smaller than denom1.
+// The pool is created with zero liquidity and the initial sqrt price and current tick set to zero.
+// The given token denominations are ordered to ensure that the first token is the numerator of the price, and the second token is the denominator of the price.
+// The pool is added to the pool store, and an error is returned if the operation fails.
 func (k Keeper) CreateNewConcentratedLiquidityPool(
 	ctx sdk.Context,
 	poolId uint64,
 	denom0, denom1 string,
-	currSqrtPrice sdk.Dec,
-	currTick sdk.Int,
+	tickSpacing uint64,
 ) (types.ConcentratedPoolExtension, error) {
+	// Order the initial pool denoms so that denom0 is lexicographically smaller than denom1.
 	denom0, denom1, err := types.OrderInitialPoolDenoms(denom0, denom1)
 	if err != nil {
 		return nil, err
 	}
-	pool := &model.Pool{
-		// TODO: move gammtypes.NewPoolAddress(poolId) to swaproutertypes
-		Address:          gammtypes.NewPoolAddress(poolId).String(),
-		Id:               poolId,
-		CurrentSqrtPrice: currSqrtPrice,
-		CurrentTick:      currTick,
-		Liquidity:        sdk.ZeroDec(),
-		Token0:           denom0,
-		Token1:           denom1,
+
+	// Create a new concentrated liquidity pool with the given parameters.
+	poolI, err := model.NewConcentratedLiquidityPool(poolId, denom0, denom1, tickSpacing)
```

Comment:
In `NewConcentratedLiquidityPool`, do have to return a `PoolI`. If not, maybe we just return the "concentrated interface extension" directly so that we don't have to do conversions?

Reply:
The other pool types return PoolI, I think we should follow the same API. Lmk what you think.

Reply:
`NewConcentratedLiquidityPool` returns a CL pool, not a pool interface.

So we should not need to cast in the first place because CL pool should implement the `ConcentratedPoolExtension`. However, it does not seem to do so which isn't right.

Reply:
We should make sure that CL pool implements `ConcentratedPoolExtension`

Reply:
It seems if I want to follow Stableswap and Balancer, the CreateNewConcentratedLiquidityPool in pool.go should actually return poolI instead of ConcentratedPoolExtension

Reply:
I'm referring to `NewConcentratedLiquidityPool` and the cast that happens after

Reply:
`NewConcentratedLiquidityPool` returns `model.Pool`. `model.Pool` implements both `swaprouter.PoolI` and `cltypes.ConcentratedPoolExtension`. So cast shouldn't' be needed

Reply:
Check the most recent commit, it seems I was able to use it directly as long as it was used with a pointer. Lmk if this satisfies the requirement 

https://github.com/osmosis-labs/osmosis/pull/3671/commits/f9b57316e24ae6f374fa062d52849adad8b1fb19